plugins/jira-dashboard-backend/src/api.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '@backstage/config';
+import fetch from 'node-fetch';
+import {
+  Project,
+  Filter,
+  Issue,
+  JiraIssue,
+  NewJiraIssueResponse,
+} from './types';
+import { JIRA_BASE_URL_CONFIG_PATH, JIRA_TOKEN_CONFIG_PATH } from './constants';
+
+export const getProjectInfo = async (
+  projectKey: string,
+  config: Config,
+): Promise<Project> => {
+  const response = await fetch(
+    `${config.getString(JIRA_BASE_URL_CONFIG_PATH)}project/${projectKey}`,
+    {
+      method: 'GET',
+      headers: {
+        Authorization: `${config.getString(JIRA_TOKEN_CONFIG_PATH)}`,
+        Accept: 'application/json',
+      },
+    },
+  );
+  if (response.status !== 200) {
+    throw Error(`${response.status}`);
+  }
+  return response.json();
+};
+
+export const getFilterById = async (
+  id: string,
+  config: Config,
+): Promise<Filter> => {
+  const response = await fetch(
+    `${config.getString(JIRA_BASE_URL_CONFIG_PATH)}filter/${id}`,
+    {
+      method: 'GET',
+      headers: {
+        Authorization: `${config.getString(JIRA_TOKEN_CONFIG_PATH)}`,
+        Accept: 'application/json',
+      },
+    },
+  );
+  if (response.status !== 200) {
+    throw Error(`${response.status}`);
+  }
+  const jsonResponse = await response.json();
+  return { name: jsonResponse.name, query: jsonResponse.jql } as Filter;
+};
+
+export const getIssuesByFilter = async (
+  projectKey: string,
+  query: string,
+  config: Config,
+): Promise<Issue[]> => {
+  const response = await fetch(
+    `${config.getString(
+      JIRA_BASE_URL_CONFIG_PATH,
+    )}search?jql=project=${projectKey} AND ${query}`,
```

Comment:
does project have to be part of this query, can we not trust the filter?
What if a team works across 2 or more projects and we wants to show these jira tables on a team page

Reply:
We often have components with the same names but in different projects, and I don't think we want to show issues from all of them. However, a great next step for this plugin (in future PR) could be support for handling multiple Jira projects per entity? And yes, the ability to add it to an entity team page! :) What do you think?

Reply:
For components I think you use getIssuesByComponent where i would agree project is needed.
My comment was related to filter-ids annotation only, I think here we can get unexpected results.

However now I do see now you use this method getIssuesByFilter with the default filter from getDefaultFilters and as they don't have project in them it's needed here. 
Maybe you could consider updating the default filters to include project?

Reply:
Ah now I see. Sure! I'll look into it, sounds like a good idea :)